### PR TITLE
Recover data from Thursday-Friday of Hormud's week 1 failure

### PR DESCRIPTION
**Pull request by as2388**
`de_identify_csv.py` is copied from IMAQAL.

The core of the updates to fetch_raw_data.py are adapted from IMAQAL, but updated to use Shaqadoon’s new recovery format and to fit into the new structure of fetch_raw-data.py.

Everything else is new.

### CODE REVIEW

File: de_identify_csv.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import argparse
```
Comment (lukechurch) [inline de_identify_csv.py:1]: I'm less fond of us copying these tools that don't get run as part of the pipeline from project to project. Could we consider putting them in infrastructure or a tools library to get a bit more standardisation and visibility of the tools that are available?

File: fetch_raw_data.py [hunk 2]
```diff
@@ -137,6 +141,54 @@ def fetch_from_gcloud_bucket(google_cloud_credentials_file_path, raw_data_dir, g
                 google_cloud_credentials_file_path, blob_url, traced_runs_output_file)
 
 
+def fetch_from_shaqadoon_csv(user, google_cloud_credentials_file_path, raw_data_dir, phone_number_uuid_table,
+                             shaqadoon_csv_source):
+    log.info("Fetching data from a Shaqadoon CSV...")
+    for blob_url in shaqadoon_csv_source.activation_flow_urls + shaqadoon_csv_source.survey_flow_urls:
+        traced_runs_output_path = f"{raw_data_dir}/{blob_url.split('/')[-1].split('.')[0]}.jsonl"
```
Comment (lukechurch) [inline fetch_raw_data.py]: An example in a comment would help legibility here

File: pipeline_config.json [hunk 3]
```diff
@@ -36,6 +36,12 @@
       "TestContactUUIDs": [
         "61e74e0b-1b4a-41fe-9e78-712f2a048f4e"
       ]
+    },
+    {
+      "SourceType": "ShaqadoonCSV",
+      "ActivationFlowURLs": [
+        "gs://avf-project-datasets/2019/OCHA/hormud-recovery-1.csv"
```
Comment (lukechurch) [inline pipeline_config.json:43]: Please could you chat with @IsaackMwenda and agree a naming scheme for files like this. We've now seen a couple of different options on IMAQAL, and I think this is also an additional options.

In general it's going to be hard to cover all of the cases, but I think we've probably seen enough examples to have a first attempt at a naming scheme.

**Review by lukechurch — APPROVED** [review]
Thanks @as2388

File: de_identify_csv.py (hunk 1 above)
Comment (as2388) [reply de_identify_csv.py:1]: I think PipelineInfrastructure is the obvious place to put them, and this script is in a position where we can do that now (this is the third project we've used it on, with the only modification being the conversion to Firestore uuid tables).

File: pipeline_config.json (hunk 3 above)
Comment (as2388) [reply pipeline_config.json:43]: Will address in the next PR (which contains Wednesday's data)

File: fetch_raw_data.py (hunk 2 above)
Comment (as2388) [reply fetch_raw_data.py]: Added a comment and moved the split code to a separate variable to also help with legibility

File: de_identify_csv.py (hunk 1 above)
Comment (as2388) [reply de_identify_csv.py:1]: Will remove in a future PR once a relevant PR in Pipeline-Infrastructure has been accepted

File: pipeline_config.json (hunk 3 above)
Comment (as2388) [reply pipeline_config.json:43]: addressed here: https://github.com/AfricasVoices/Project-OCHA/pull/47

**Pull request merged**